### PR TITLE
FSM produces force progress

### LINKED ISSUE #3127: FSM produces on-chain forced progress tx
In order for participants to be safe, one must be able to produce a `channel_force_progress_tx` that lands on-chain.

Provide an API to the FSM that is closely resembling the channel contract call one but that produces an `channel_force_progress_tx` instead. Optionally provide fee, gas_price, and nonce

### PR DESCRIPTION
**Pull request by velzevur**
This PR introduces FSM functionality to produce force progress transactions.
Closes #3127 

It also fixes a few bugs found along the way.

### CODE REVIEW

**Comment by velzevur** [comment]
What is left to do is make the FP optional variables adjustable: namely `nonce`, `gas_price` and `fee`

File: apps/aechannel/src/aesc_fsm.erl [hunk 1]
```diff
@@ -1814,21 +1852,52 @@ new_onchain_tx_(Mod, Opts, CurrHeight) when Mod =:= aesc_create_tx;
...
+    GasPrice = MinFee div FeeGas,
     case MinFee =< TxFee of
         true ->
-            {ok, Tx};
+            case Mod of
+                aesc_force_progress_tx ->
+                    {aesc_force_progress_tx, FPTx} =
+                        aetx:specialize_callback(Tx),
+                    [Update] = aesc_force_progress_tx:updates(FPTx),                    
+                    %DOMAT
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: ?

File: apps/aechannel/src/aesc_fsm.erl [hunk 2]
```diff
@@ -1814,21 +1852,52 @@ new_onchain_tx_(Mod, Opts, CurrHeight) when Mod =:= aesc_create_tx;
...
+                    case aesc_offchain_update:get_gas_price(Update) of
+                        {ok, UpdateGasPrice} when UpdateGasPrice >= GasPrice -> %% same gas_price
+                            {ok, Tx};
+                        {ok, _OtherGasPrice} ->
+                            error(deletemeplease),
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: ?

File: apps/aechannel/src/aesc_fsm.erl [hunk 3]
```diff
@@ -4945,3 +5087,60 @@ mutual_close_error(Msg, D) ->
...
+
+mk_force_progress_tx(Opts, #data{state = State, opts = ChannelOpts,
+                                 on_chain_id = ChannelPubkey}) ->
+    CallStack = maps:get(call_stack, Opts, []),
+    #{ contract    := ContractPubKey
+     , abi_version := ABIVersion
+     , amount      := Amount
+     , call_data   := CallData
+     , acct        := FromPub } = Opts,
+    GasPrice = maps:get(gas_price, Opts, 1000000000),
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: Should these defaults really be hard-coded here?

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 4]
```diff
@@ -4389,8 +4423,273 @@ settle_with_failed_onchain(Cfg) ->
...
+    CreateArgs = #{ vm_version  => aect_test_utils:vm_version()
+                  , abi_version => aect_test_utils:abi_version() 
+                  , deposit     => Deposit
+                  , code        => BinCode
+                  , call_data   => CallData},
+    aesc_fsm:upd_create_contract(FsmC, CreateArgs),
+    {Creator1, _} = await_signing_request(update, Creator, Cfg),
+    await_update_incoming_report(Acknowledger, ?TIMEOUT, Debug),
+    {Acknowledger1, _} = await_signing_request(update_ack, Acknowledger,
+                                              Cfg),
```
Comment (uwiger) [inline apps/aechannel/test/aesc_fsm_SUITE.erl]: Why a line break?

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 5]
```diff
@@ -4389,8 +4423,273 @@ settle_with_failed_onchain(Cfg) ->
...
+                                         FunName, FunArgs),
+    CallArgs = #{ contract    => ContractId
+                , abi_version => aect_test_utils:abi_version() 
+                , amount      => Amount
+                , call_data   => CallData},
+    aesc_fsm:upd_call_contract(FsmC, CallArgs),
+    {Caller1, _} = await_signing_request(update, Caller, Cfg),
+    await_update_incoming_report(Acknowledger, ?TIMEOUT, Debug),
+    {Acknowledger1, _} = await_signing_request(update_ack, Acknowledger,
+                                              Cfg),
```
Comment (uwiger) [inline apps/aechannel/test/aesc_fsm_SUITE.erl]: Why a line break?

**Pull request merged**